templates/publisher/listing.html

```diff
@@ -95,19 +95,31 @@
 
         <div class="row" data-validation-name="icon">
           <div class="col-2">
-            <div class="p-media-object__image--large p-editable-icon">
-              <img id="snap_icon_image"
-                {% if icon_url %}
-                  src="{{ icon_url }}" alt="{% if display_title %}{{ display_title }}{% else %}{{ snap_title }}{% endif %} snap"
-                {% else %}
-                  src="https://assets.ubuntu.com/v1/be6eb412-snapcraft-missing-icon.svg" alt=""
-                {% endif %}
-              />
+            <label class="p-editable-icon__label">Snap icon:</label>
```

Comment:
Could we link this label with icon input so clicking on it would open file dialog?